### ISSUE TITLE
Model.url() doesn't match documentation

### ISSUE THREAD

**Issue by deianvasilev** [jashkenas/backbone#1245]
According Backbone.js documentation:
model.ur() Generates URLs of the form: "/[collection.url]/[id]", falling back to "/[urlRoot]/id" if the model is not part of a collection

so this line in the code:

var base = getValue(this, 'urlRoot') || getValue(this.collection, 'url') || urlError();

should look like this:

var base = getValue(this.collection, 'url') || getValue(this, 'urlRoot') || urlError();

Basically at the moment if urlRoot is set on model, model never checks if he "belong" to collection
